docs: Make supported language explicit (#1313)
grafana/alloy: docs/sources/reference/components/pyroscope/pyroscope.ebpf.md

You can specify multiple `pyroscope.ebpf` components by giving them different labels, however it's not recommended as it can lead to additional memory and CPU usage.

## Supported languages

> This eBPF profiler only collects CPU profiles. Generally natively compiled languages like C/C++, Go, Rust are supported (also see [Troubleshooting]({{< relref "#troubleshooting" >}}) for additional requirements).

Reviewer: ```suggestion
This eBPF profiler only collects CPU profiles. Generally, natively compiled languages like C/C++, Go, and Rust are supported. Refer to [Troubleshooting unknown symbols](#troubleshooting-unknown-symbols) for additional requirements.
```

relrefs are deprecated. We should use inline links or define the links at the bottom of the section or page.

Author: Do you have an example by hand?

Reviewer: Inline linking is like I suggested here.

Links at the bottom of the section/page is used in most topics - for example in `docs/sources/get-started/configuration-syntax/components.md` (we can use relative links or fully qualified URLs.

Author: I see, sorry I wasn't properly reading the suggestion.

> Python is the only supported high-level language, as long as `python_enabled=true`. Other high-level languages like Java, Ruby, PHP, and JavaScript require additional work to show stack traces of methods in these languages correctly. Currently, their CPU usage will be reported belonging to the runtime's methods instead.

Reviewer: ```suggestion
Python is the only supported high-level language, as long as `python_enabled=true`.
Other high-level languages like Java, Ruby, PHP, and JavaScript require additional work to show stack traces of methods in these languages correctly.
Currently, the CPU usage for these languages is reported as belonging to the runtime's methods.
```

> The only high-level language supported is Python (as long as `python_enabled=true`). Other high-level languages like Java, Ruby, PHP, JavaScript, etc. will require some additional work in order to correctly show stack traces of methods in these languages. Currently their CPU usage will be displayed belonging to the runtime's methods instead.

Reviewer: ```suggestion
Python is the only supported high-level language, as long as `python_enabled=true`. Other high-level languages like Java, Ruby, PHP, and JavaScript require additional work to show stack traces of methods in these languages correctly. Currently, their CPU usage will be displayed belonging to the runtime's methods instead.
```

Where is the CPU usage displayed? Who is displaying it? Alloy? Grafana?

Author: I guess Grafana is displaying it in the flamegraphs, maybe we us reported, as that is what Alloy is doing?

```suggestion
Python is the only supported high-level language, as long as `python_enabled=true`. Other high-level languages like Java, Ruby, PHP, and JavaScript require additional work to show stack traces of methods in these languages correctly. Currently, their CPU usage will be reported belonging to the runtime's methods instead.
```

If `service_name` isn't specified and couldn't be inferred, it's set to `unspecified`.

> ## Troubleshooting unknown symbols {#troubleshooting}

Reviewer: ```suggestion
## Troubleshooting unknown symbols
```

We don't need this if the link is defined as suggested above.

- The `.symtab` and `.dynsym` sections in the debug ELF file.
- The `.gopclntab` section in Go language ELF files.

> The search for debug files follows [gdb algorithm].

Reviewer: ```suggestion
The search for debug files follows [gdb algorithm][].
```

We can add the extra `[]` just for consistency. It's not required but we tend to use it throughout the rest of the docs.
